v3/state.py

```diff
@@ -131,6 +131,19 @@ def readFromMemoryOrDisk(self, data, data_path, save=False):
                     .with_columns(
                         as_of=pl.col("block_number") + pl.col("transaction_index") / 1e4
                     )
+                    .cast(
+                        {
+                            "block_timestamp":pl.Datetime,
+                            "amount0":pl.Float64,
+                            "amount1":pl.Float64,
+                            "sqrtPriceX96":pl.Float64,
+                            "liquidity":pl.Float64,
```

Comment:
might be nice to see if this is convertible to a no precision loss value since liquidity is must be less than an int128, but in practice is quite a bit smaller than that (due to the max liquidity in tick math)